packages/vscode-extension/lib/expo_router_v2_plugin.js

```diff
@@ -33,8 +54,12 @@ function useRouterPluginMainHook({ onNavigationChange }) {
         id: computeRouteIdentifier(snapshot.pathname, snapshot.params),
       };
     },
-    requestNavigationChange: ({ pathname, params }) => {
-      router.push(pathname, params);
+    requestNavigationChange: (navigationDescriptor) => {
```

Comment:
@jakub-gonet Maybe we could simplify this approach and use while loop instead? What do you think?
```tsx
while (store.navigationRef?.isReady() === false) {
        await new Promise((resolve) => setTimeout(resolve, 100));
      }
      requestNavigationChange(navigationDescriptor);
```